prybar_assets/nodejs/module-context-hook.js

```diff
@@ -193,7 +193,15 @@ function runModule(moduleName, isInteractive) {
     });
   }
 
-  require(absPath);
+  try {
+    require(absPath);
+  } catch (e) {
+    if (e.code === 'ERR_REQUIRE_ESM') {
+      import(absPath);
+    } else {
+      throw e;
```

Comment:
what happens if we always import instead?

Reply:
I don’t think there is any difference, it’s just safer to do this for now.

Reply:
this will break the repl's context and I don't think it'll help much in repls since in normal node you can only have one or the other.

Instead what I'd suggest doing is to use source maps and esbuild to transform esm code to cjs.  It's slower but I think its our best option, as much as I don't like it.

Reply:
what does it mean to break the repl's context?

Reply:
Capital REPL not repl. Haha

Basically having access to the module scope inside the console after running.

My favorite Replit feature, patching an express server while it's running.


https://user-images.githubusercontent.com/1994028/165374690-215f96a1-b234-425c-8420-03980686e5ab.mp4

or simply

![image](https://user-images.githubusercontent.com/1994028/165374941-fc3f37b6-bd87-47aa-929b-b8cedcfafa03.png)


surprisingly, for my previous user research beginners find this intuitive, but professionals are surprised it works.

Reply:
> surprisingly, for my previous user research beginners find this intuitive, but professionals are surprised it works

i love this